src/Microsoft.AspNetCore.Mvc.RazorPages/RazorPagesOptions.cs

```diff
@@ -42,5 +42,41 @@ public string RootDirectory
                 _root = value;
             }
         }
+
+        /// <summary>
+        /// Gets or sets a value that determines if areas are enabled for Razor Pages.
+        /// Defaults to <c>true</c>.
+        /// <para>
+        /// When enabled, any Razor Page under the directory structure <c>/{AreaRootDirectory}/AreaName/{RootDirectory}/</c>
+        /// will be associated with an area with the name <c>AreaName</c>.
+        /// <seealso cref="AreaRootDirectory"/>
+        /// <seealso cref="RootDirectory"/>
+        /// </para>
+        /// </summary>
+        public bool EnableAreas { get; set; } = true;
```

Comment:
How did we decide on `true`? That's a breaking change

Reply:
I personally think it's a super minor breaking change. I could certainly be swayed about the default value depending on what we do regarding Compat Switches/Levels. But if this is off by default for at least *new* projects (created from templates), that would be a major usability problem. But I think I want it on even for "legacy" projects. I feel that the only scenario it breaks is something incredibly obscure.

Reply:
Actually I slightly take this back. I had confused myself over what scenario it breaks. It breaks the scenario where you set the Pages Root to be `/`, and I actually have no sense of whether that is common or not. I *hope* it's very uncommon, because it strikes me as a very dangerous thing to do. Razor Pages will find all kinds of crazy CSHTML files the developer left laying around their app (including MVC views!). So, it would easily break someone who did something I consider to be odd, but maybe that's ok?

Reply:
> I could certainly be swayed about the default value depending on what we do regarding Compat Switches/Levels. But if this is off by default for at least new projects 

The general principle of the compat switches design I've been working on is that new behaviours are on new projects created with the template.

The reason why I wanted to get compat switches into 2.1 is that it makes our decision process very mechanical from now on. I think we've got 5 of these things (changes in a shipped feature).

Reply:
Agreed.

Reply:
Fair enough. If we do leave this not-enabled, @javiercn would have to set it as part of the Identity setup.

Reply:
Moving the comment above to here:

I chatted with @rynowak. While I agree this is a breaking change (as in a behavior change), it relies on two requirements that I don't think would be too common to happen at the same time:
* Change the pages root to be the content root.
* **Have an existing Areas folder on your application with pages inside it.**

There are a few options here:

* Do the "breaking change" and not provide any option.
* Do the "breaking change" by default and provide the ability to turn it off.
* Do not do the "breaking change" and provide the ability to turn the behavior on with the new compatibility switches functionality.

I would like to suggest that for preview 1 we do the feature on by default and without providing any option/compatibility switch for it, make an announcement on the announcements repository and gather feedback from people in order to make a decision that, absent data, adds complexity to this feature with limited value (based on the assumptions made above, we don't expect people to be broken by this)

Reply:
Discussed @DamianEdwards and @Eilon - the plan is to make this off by default.

Reply:
Tracked via https://github.com/aspnet/Mvc/issues/7148